scripts/ci/pre_commit/pre_commit_inline_scripts_in_docker.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from os import listdir
+from pathlib import Path
+from typing import List
+
+AIRFLOW_SOURCES_DIR = Path(__file__).parent.parent.parent.parent
+
+
+def insert_content(file_path: Path, content: List[str], header: str, footer: str, file_name: str):
+    text = file_path.read_text().splitlines(keepends=True)
+    replacing = False
+    result: List[str] = []
+    for line in text:
+        if line.startswith(f"{header}{file_name}"):
+            replacing = True
+            result.append(line)
+            result += content
+        if line.startswith(footer):
+            replacing = False
+        if not replacing:
+            result.append(line)
+        file_path.write_text("".join(result))
+
+
+if __name__ == '__main__':
+    DOCKERFILE_FILE = AIRFLOW_SOURCES_DIR / 'Dockerfile'
+    DOCKERFILE_CI_FILE = AIRFLOW_SOURCES_DIR / 'Dockerfile.ci'
+    SCRIPTS_DOCKER_DIR = AIRFLOW_SOURCES_DIR / 'scripts' / 'docker'
+
+    for file in [DOCKERFILE_FILE, DOCKERFILE_CI_FILE]:
+        for script in listdir(SCRIPTS_DOCKER_DIR):
+            script_content = (SCRIPTS_DOCKER_DIR / script).read_text().splitlines(keepends=True)
+            no_comments_script_content = [
+                line for line in script_content if not line.startswith("#") or line.startswith("#!")
```

Comment:
Removing comments make the generated Dockerfile content smaller, but I am checking with legal if this not too much (the scripts inside the Dockerfile will have licence information stripped off).

Reply:
I started discussion in https://lists.apache.org/thread/jjhfk95ch12qskynlvxqzd9n77o7152r